src/Http/ServerRequest.php

```diff
@@ -1722,7 +1722,7 @@ public function getUri(): UriInterface
      * @return static
      * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
```

Comment:
Was this missing for a reason due to a base class? We added a suppress for it.

Reply:
Maybe back in the days, now I dont see why
`$new->uri` already has the typehint, so it couldnt possibly be a different type.

Reply:
Ah, so apparently the parent does not have bool type, thats why the annotation is needed

`Method \Cake\Http\ServerRequest::withUri() does not have native type hint for its parameter $preserveHost but it should be possible to add it based on @param annotation "bool".` must be ignored to prevent issues.

Reply:
Seems the missing type for `$uri` was an oversight. It's the `$preserveHost` are for which we cannot add the type, to match the interface signature https://github.com/php-fig/http-message/blob/efd67d1dc14a7ef4fc4e518e7dee91c271d524e4/src/RequestInterface.php#L128.

We'll have to unfortunately carry this wackiness since PHP-FiG don't seem to be interested in making new releases for th `http-message` package with proper types added.